user-guide/Troubleshooting/Procedures/SLScripting_FinalizerException.md

```diff
@@ -0,0 +1,258 @@
+
+## Introduction
+
+A [finalizer](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/finalizers) can be used to perform cleanup when the class instance is being collected by the garbage collector. Instances of a class that contain a Finalizer get added to the Finalize queue which then get processed by the finalizer thread.
+It's important to make sure the Finalizer does not throw an exception as this results in a process crash when the Finalizer thread is executing the Finalizer method of the type that is being freed.
+
+> [!NOTE]
+>
+> - Typically, you can avoid having to provide a finalizer by using the [SafeHandle](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle) or derived classes to wrap any unmanaged handle.
+> - Ift is recommended to provide a way to explicitly release resource before the garbage collector frees the object by implementing the [IDisposable](https://learn.microsoft.com/en-us/dotnet/api/system.idisposable?view=netframework-4.8.1) interface to perform the necessary cleanup for the object. This can considerably improve the performance of the application. Note however that even if you implement IDisposable, the finalizer acts as a safeguard to clean up the resources if the call to the Dispose method fails.
+
+For more information about cleaning up resources, refer to:
+
+- [Cleaning Up Unmanaged Resources](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/unmanaged)
+- [Implementing a Dispose Method](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose)
+- [Implementing a DisposeAsync Method](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync)
+- [using statement](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/using)
+
+## Visual Studio
+
+In case a crash occurred due to an exception in the Finalize method of a class instance, a dump of SLManagedScripting should have been created.
+The ERRORLOG.TXT file should contain an indication of this exception. For example:
+
+```txt
+System.NullReferenceException: Object reference not set to an instance of an object.
+   at Skyline.Protocol.MyExtension.MyClass.Finalize()
+```
+
+Open the dump file in Visual Studio, an press the *Run Diagnostic Analysis* link in the Actions list.
+The Analysis results pane should show that an exception was found on the heap:
+
+![Analysis results](~/user-guide/images/VisualStudioDiagnosticAnalysFinalizerCrash.png)<br>*Visual Studio Analysis results*
+
+In this case it mentions that a [NullReferenceException](https://learn.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=netframework-4.8.1) was found.
+
+Now open the Parallel Stacks window (*Debug* > *Windows* > *Parallel Stacks*):
+
+![WinDbg](~/user-guide/images/VisualStudioParallelStacksWindowFinalizerCrash.png)<br>*Visual Studio Parallel Stacks window*
```

Comment:
Is WinDbg the correct image placeholder here?
Additionally, I would consider highlighting the thread in the Parallel Stacks

Reply:
updated and added highlight